sdks/python/apache_beam/io/gcp/bigquery_tools.py

```diff
@@ -347,16 +347,21 @@ def _start_query_job(self, project_id, query, use_legacy_sql, flatten_results,
             jobReference=reference))
 
     response = self.client.jobs.Insert(request)
-    return response.jobReference.jobId
+    return response.jobReference.jobId, response.jobReference.location
 
   @retry.with_exponential_backoff(
       num_retries=MAX_RETRIES,
       retry_filter=retry.retry_on_server_errors_and_timeout_filter)
   def _get_query_results(self, project_id, job_id,
-                         page_token=None, max_results=10000):
-    request = bigquery.BigqueryJobsGetQueryResultsRequest(
-        jobId=job_id, pageToken=page_token, projectId=project_id,
-        maxResults=max_results)
+                         page_token=None, max_results=10000, location=None):
+    if location:
+      request = bigquery.BigqueryJobsGetQueryResultsRequest(
+          jobId=job_id, pageToken=page_token, projectId=project_id,
+          maxResults=max_results, location=location)
+    else:
+      request = bigquery.BigqueryJobsGetQueryResultsRequest(
+          jobId=job_id, pageToken=page_token, projectId=project_id,
+          maxResults=max_results)
```

Comment:
I believe it's okay to do location=None as well, so you don't need to have the `if`/`else` path here.

Reply:
@pabloem  Yeah you are right. Thanks for the suggestion!